### DISCUSSION TITLE
My folder is not tracked by git

### DISCUSSION THREAD

**Discussion by tbhaxor** [community/community#22205] [New to GitHub]
<p>I have a folder <code>FBA_Matting</code>, i originally cloned the repo in the existing git repo and due to which i was unable to commit the folder.</p>
<p>So i deleted the <code>FBA_Matting/.git</code> but unfortunately the folder was not committed.</p>
<p>Then on my other PC, there was git repo containing all the code of FBA_Matting with some changes, So to avoid <code>fatal: refusing to merge unrelated histories</code></p>
<p>So i copied the folder and pasted into my repository (checkout new branch, added FBA_Matting). Now I can’t see my folder back in <code>git status</code> command. So i tried to change the name from <code>FBA_Matting</code> to <code>FBAMatting</code> and  now i can see it in <code>git status</code></p>
<p>PS: I checked <code>.gitignore</code> Also, and there as no file containing FBA_Matting entry</p>

**Comment by ernest-phillips** [comment]
<p>Hi <a class="mention" href="/tbhaxor">@tbhaxor</a></p>
<p>Are you still having trouble tracking that folder in Git?<br>
How did you add the folder? Please give steps.</p>
<p>Was it added locally in your file directory? Or was it uploaded to the GitHub UI?<br>
Please list all the git commands used to get to the point where you are now.<br>
This will help the community reproduce what you are experiencing.</p>

**Comment by tbhaxor** [comment]
<p>Not actually, I just want to know this</p>
<p>In git repo folder i cloned another git repo (FBA_Matting). All the folders we committed but not FBA_Matting. Then later on when i removed .git folder from FBA_Matting, the contents in folder was lost.  There are empty folder committed.</p>
<p>The problem is Git never commit empty folder. How can i retrieve my files of FBA_Matting.</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>You can check your logs to go back to a previous point in time.<br>
The following steps assume you deleted the contents and then made commits after.<br>
To find the right commit, first check the history for the deleted file:</p>
<pre><code class="lang-auto">$ git log -- &lt;filename&gt;
</code></pre>
<p>You can either work with the last commit that still had the file, or the commit that deleted the file. In the first case, just checkout the file from that commit:</p>
<pre><code class="lang-auto">$ git checkout &lt;commit hash&gt; -- &lt;filename&gt;
</code></pre>
<p>In the second case, checkout the file from one commit <em>before</em> that:</p>
<pre><code class="lang-auto">$ git checkout &lt;deletion commit hash&gt;~1 -- &lt;filename&gt;
</code></pre>
<p>If the file was never actually committed this of course will not be in the history.</p>
<p>You mentioned:</p>
<blockquote>
<p>Then on my other PC, there was git repo containing all the code of FBA_Matting with some changes, So to avoid fatal: refusing to merge unrelated histories</p>
</blockquote>
<p>With the folder added this changes the history, so you will need to address the differences between the files and fix the merge conflicts.</p>
<p>Use :<br>
$ git pull origin [repo] --allow-unrelated-histories</p>
<p>‍</p>
